pkg/apis/options/providers.go

```diff
@@ -0,0 +1,172 @@
+	ProtectedResource string `json:"resource,omitempty"`
+	// ValidateURL is the access token validation endpoint
+	ValidateURL string `json:"validateURL,omitempty"`
+	// Scope is the OAuth scope specification
+	Scope string `json:"scope,omitempty"`
+	// Prompt is OIDC prompt
+	Prompt string `json:"prompt,omitempty"`
+	// ApprovalPrompt is the OAuth approval_prompt
+	// default is set to 'force'
+	ApprovalPrompt string `json:"approvalPrompt,omitempty"`
+	// AllowedGroups is a list of restrict logins to members of this group
+	AllowedGroups []string `json:"allowedGroups,omitempty"`
+
+	// AcrValues is a string of acr values
+	AcrValues string `json:"acrValues,omitempty"`
+}
+
+type KeycloakOptions struct {
+	//KeycloakGroup enables to restrict login to members of indicated group
+	KeycloakGroup string `json:"keycloakGroup,omitempty"`
+}
+
+type AzureOptions struct {
+	// AzureTenant directs to a tenant-specific or common (tenant-independent) endpoint
+	// Default value is 'commmon'
+	AzureTenant string `json:"azureTenant,omitempty"`
+}
+
+type BitbucketOptions struct {
+	// BitbucketTeam sets restrict logins to members of this team
+	BitbucketTeam string `json:"bitbucketTeam,omitempty"`
+	// BitbucketRepository sets restrict logins to user with access to this repository
+	BitbucketRepository string `json:"bitbucketRepository,omitempty"`
+}
+
+type GitHubOptions struct {
+	// GitHubOrg sets restrict logins to members of this organisation
+	GitHubOrg string `json:"githubOrg,omitempty"`
+	// GitHubTeam sets restrict logins to members of this team
+	GitHubTeam string `json:"githubTeam,omitempty"`
+	// GitHubRepo sets restrict logins to collaborators of this repository
+	GitHubRepo string `json:"githubRepo,omitempty"`
+	// GitHubToken is the token to use when verifying repository collaborators
+	// it must have push access to the repository
+	GitHubToken string `json:"githubToken,omitempty"`
+	// GitHubUsers allows users with these usernames to login
+	// even if they do not belong to the specified org and team or collaborators
+	GitHubUsers []string `json:"githubUsers,omitempty"`
+}
+
+type GitLabOptions struct {
+	// GitLabGroup sets restrict logins to members of this group
+	GitLabGroup []string `json:"gitlabGroups,omitempty"`
+}
+
+type GoogleOptions struct {
+	// GoogleGroups sets restrict logins to members of this google group
+	GoogleGroups []string `json:"googleGroup,omitempty"`
+	// GoogleAdminEmail is the google admin to impersonate for api calls
+	GoogleAdminEmail string `json:"googleAdminEmail,omitempty"`
+	// GoogleServiceAccountJSON is the path to the service account json credentials
+	GoogleServiceAccountJSON string `json:"googleServiceAccountJson,omitempty"`
+}
+
+type OIDCOptions struct {
+	// OIDCIssuerURL is the OpenID Connect issuer URL
+	// ie: https://accounts.google.com
+	OIDCIssuerURL string `json:"oidcIssuerURL,omitempty"`
+	// InsecureOIDCAllowUnverifiedEmail preventes failur if an email address in an id_token is not verified
```

Comment:
Do you know if this was a copy paste, there's a typo here, might be worth fixing up in the original location if it was copy/pasted
```suggestion
	// InsecureOIDCAllowUnverifiedEmail prevents failures if an email address in an id_token is not verified
```

Reply:
nope, all good in the original bcs I did a paraphrase here..